plugins/chat/assets/javascripts/discourse/services/chat-channel-thread-pane.js

```diff
@@ -26,19 +26,12 @@ export default class ChatChannelThreadPane extends ChatChannelPane {
   }
 
   get lastCurrentUserMessage() {
-    const lastCurrentUserMessage =
-      this.chat.activeChannel.activeThread.messages.findLast(
-        (message) => message.user.id === this.currentUser.id
-      );
-
-    if (!lastCurrentUserMessage) {
-      return;
-    }
-
-    if (lastCurrentUserMessage.staged || lastCurrentUserMessage.error) {
-      return;
-    }
-
-    return lastCurrentUserMessage;
+    return this.chat.activeChannel.activeThread.messages.findLast(
```

Comment:
I feel like we could move this whole `lastCurrentUserMessage` method into `messagesManager`?  We already do `setOwner` in there so we could look up the current user service...or just pass in the `currentUser` like:

```js
this.chat.activeChannel.activeThread.messages.findLastForCurrentUser(currentUser)
```

Reply:
Yes good idea, I refactored it a little bit like this